accepted/build/references-into-frameworkreferences.md

```diff
@@ -0,0 +1,59 @@
+
+With this proposal, we would not reference any of the theme assemblies by default.  To use a single theme assembly, you would add a reference to it:
+
+```
+<Reference Include="PresentationFramework.AeroLite" />
+```
+
+To reference multiple theme assemblies, you would add references with aliases:
+
+```
+<Reference Include="PresentationFramework.AeroLite" Aliases="AeroLite" />
+<Reference Include="PresentationFramework.Luna" Aliases="Luna" />
+```
+
+More discussion of the WPF Theme assemblies (and the inspiration for this proposal) is here: https://github.com/dotnet/sdk/issues/3265
+
+### Turn off default references to assemblies in targeting pack
+
+- Projects in CoreFx [would like](https://github.com/dotnet/sdk/issues/3295) to have a `FrameworkReference` to Microsoft.NETCore.App in order to get the App Host and the right data in the runtimeconfig.json file.  However, they can't include the normal references from the targeting pack, as they are building projects which are part of the targeting pack.
+- WPF wants to explicitly choose all their references, in order to avoid accidentally taking a dependency on APIs such as LINQ.
+- Libraries might want to reference specific assemblies in a shared framework without exposing the whole shared framework as API surface to their consumers.
+
+The project would include something like this:
+
+```xml
+<FrameworkReference Update="Microsoft.WindowsDesktop.App" IncludeReferences="False" />
+<Reference Include="System.Printing" />
+```
+
+## Implementation details
+
+### ResolveAssemblyReference
+
+The ResolveAssemblyReference task would be updated to understand the "new-style" targeting packs, and the FrameworkList files in them.  There would be an additional parameter to the task to pass in the resolved targeting packs and paths.  The SearchPaths parameter would also support an additional search path for the targeting packs.  There would be an additional resolver used by ResolveAssemblyReference which would resolve simple name References to the reference assemblies in the targeting pack.
+
+### Conflict Resolution
+
+Conflict resolution runs before ResolveAssemblyReference, however there may be conflicts involving references that will be resolved from a targeting pack by ResolveAssemblyReference.  If there is a simple name reference to an assembly which is in multiple targeting packs, then ResolveAssemblyReference needs to pick the best one (using similar logic to conflict resolution).  If there is a conflict between a simple name which will be resolved by ResolveAssemblyReference and another reference (either from a NuGet package or that has been resolved from a targeting pack already by ResolveTargetingPackAssets), then ResolvePackageFileConflicts needs to resolve the conflict.  It already does something similar for .NET Framework, by reading the FrameworkList to get the version information that a reference will resolve to.
```

Comment:
Will conflict resolution handle `ProjectReference` vs. another reference correctly?

Reply:
https://github.com/dotnet/sdk/issues/3254

Reply:
Do you mean if you're referencing a project that produces the same assembly that's in a targeting pack?  If so, I wouldn't guarantee that conflict resolution will correctly handle that by default, as that should only happen for projects that are included in the targeting pack, which may have to add special case logic for this.

Reply:
Yes, that's exactly what I mean. `Microsoft.NetCore.App` contains `WindowsBase.dll`, and if WPF repo tries to build `PresentationFramework.csproj` with a reference to `WindowsBase.csproj`, and a `FrameworkReference` to `Microsoft.NetCore.App`, will conflict resolution ensure that `WindowsBase.csproj` wins (alongwith the correct results in `deps.json` etc.)? 

If that's not the case (I'm reasonably certain this doesn't work today - we have had to work aroudn this in WPF repos), can this scenario be fixed in conflict-resolution as part of this work?